src/python/pants/backend/shell/target_types.py

```diff
@@ -442,6 +461,41 @@ class ShellCommandTarget(Target):
     )
 
 
+class ShellRunInSandboxTarget(Target):
+    alias = "experimental_run_in_sandbox"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        RunInSandboxRunnableField,
+        RunInSandboxArgumentsField,
+        ShellCommandExecutionDependenciesField,
+        ShellCommandOutputDependenciesField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
+        RunInSandboxSourcesField,
+        ShellCommandTimeoutField,
+        ShellCommandToolsField,
+        ShellCommandExtraEnvVarsField,
+        EnvironmentField,
+    )
+    help = softwrap(
+        """
+        Execute any runnable target for its side effects.
+
+        Example BUILD file:
+
+            experimental_run_in_sandbox(
+                runnable=":python_source",
```

Comment:
Can this be a path to a file (e.g. `./whatever.py`)?

I imagine that'd be the most convenient way to invoke this in most cases, since I imagine most Python files get loaded into pants via `python_sources()`, and thus don't have an obvious `:...` target name that uniquely identifies just one file?

Reply:
Yes: as long as `whatever.py` is declared as a `python_source` in some way (e.g. by `python_sources()`). Any valid target identifier is usable here, and for `python_source` targets, the file name is often correct.

The intention is to make a LOT more things runnable, e.g. `python_requirement`, so you could pull down a version of a requirement from PyPI and run that in Pants (if it exposes `python -m` functionality). Such targets do not correspond to files on disk, so you'd need to specify the target identifier there instead.